android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -144,25 +144,14 @@ protected boolean hasScreen(ScreenFragment screenFragment) {
   @Override
   protected void performUpdate() {
 
+    ArrayList<ScreenStackFragment> nonDismissedScreens = getNonDismissedScreens();
```

Comment:
looks like an overkill to allocate and compute a whole array just to get the last element, maybe we can move it to a separate method but keep the previous loop here. Esp that we also make yet another call that allocates yet another array a few lines below